### PR TITLE
0.5 Support

### LINKED ISSUE #5: Support for Dioxus 0.5
Prepare support for Dioxus 0.5

For now, build on top of https://github.com/DioxusLabs/dioxus/pull/1791

### PR DESCRIPTION
**Pull request by DogeDark**
Updates ``dioxus-charts`` to use Dioxus 0.5

Closes #5

### CODE REVIEW

**Comment by DogeDark** [comment]
It's not the cleanest it could be (and there are probably tons of better ways to do stuff) but it has support for 0.5 now.

**Comment by DogeDark** [comment]
Can't find a request review button so I'll do this: @marc2332

File: examples/demo/src/lib.rs [hunk 1]
```diff
@@ -31,7 +31,7 @@ pub fn demo_element<'a>(cx: Scope<'a>) -> Element<'a> {
                                 padding_bottom: 30,
                                 bar_width: "10%",
                                 horizontal_bars: true,
-                                label_interpolation: |v| format!("{v}%"),
+                                label_interpolation: (|v| format!("{v}%")) as fn(f32) -> String,
```
Comment (marc2332) [inline examples/demo/src/lib.rs:34]: Can't you just do?:
```suggestion
                                label_interpolation: |v: f32| format!("{v}%"),
```

Comment (marc2332) [reply examples/demo/src/lib.rs:34]: Same for the test

File: src/charts/bar.rs [hunk 2]
```diff
@@ -301,192 +301,220 @@ pub fn BarChart<'a>(cx: Scope<'a, BarChartProps<'a>>) -> Element {
...
+                            stroke: "rgb({color_var}, 40, 40)",
+                            stroke_width: "{props.bar_width}",
+                        }
+                    }
+                }
+            });
+        }
+    }
+
+    println!("STUCK");*/
```
Comment (marc2332) [inline src/charts/bar.rs]: hahaa, what's this about?

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: src/charts/bar.rs (hunk 2 above)
Comment (DogeDark) [reply src/charts/bar.rs]: Oh oops. I do that when trying to rewrite something but want to keep the original as a temporary reference.

File: examples/demo/src/lib.rs (hunk 1 above)
Comment (DogeDark) [reply examples/demo/src/lib.rs:34]: Unfortunately not. According to JK, it's a Rust issue. See [here](https://discord.com/channels/899851952891002890/943190605067079712/1219870965190758482)

File: Cargo.toml [hunk 3]
```diff
@@ -11,7 +11,7 @@ categories = ["wasm", "gui", "web-programming"]
 
 [dependencies]
 log = "0.4"
-dioxus = { version = "0.4.0", default-features = false, features = ["macro", "html"] }
+dioxus = { version = "0.5.0-alpha.2", default-features = false, features = ["macro", "html"] }
```
Comment (marc2332) [inline Cargo.toml]: let's bump it to stable!
```suggestion
dioxus = { version = "0.5", default-features = false, features = ["macro", "html"] }
```

File: examples/demo/Cargo.toml [hunk 4]
```diff
@@ -8,6 +8,6 @@ license = "MIT OR Apache-2.0"
 publish = false
 
 [dependencies]
-dioxus = { version = "0.4.0", default-features = false, features = ["macro", "html"] }
+dioxus = { version = "0.5.0-alpha.2", default-features = false, features = ["macro", "html"] }
```
Comment (marc2332) [inline examples/demo/Cargo.toml]:
```suggestion
dioxus = { version = "0.5", default-features = false, features = ["macro", "html"] }
```

File: examples/web/Cargo.toml [hunk 5]
```diff
@@ -10,5 +10,5 @@ publish = false
 [dependencies]
 log = "0.4"
 env_logger = "0.10"
-dioxus-web = "0.4.0"
+dioxus = { version = "0.5.0-alpha.2", features = ["web"] }
```
Comment (marc2332) [inline examples/web/Cargo.toml]:
```suggestion
dioxus = { version = "0.5", features = ["web"] }
```

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: Cargo.toml [hunk 6]
```diff
@@ -11,7 +11,7 @@ categories = ["wasm", "gui", "web-programming"]
 
 [dependencies]
 log = "0.4"
-dioxus = { version = "0.4.0", default-features = false, features = ["macro", "html"] }
+dioxus = { version = "0.5", default-features = false, features = ["launch", "macro", "html"] }
```
Comment (marc2332) [inline Cargo.toml:14]: Do we need to add the launch feature in the library?
```suggestion
dioxus = { version = "0.5", default-features = false, features = ["macro", "html"] }
```

**Review by marc2332 — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 6 above)
Comment (DogeDark) [reply Cargo.toml:14]: See https://github.com/DioxusLabs/dioxus/issues/2162

Comment (marc2332) [reply Cargo.toml:14]: Damn, alright. Opened https://github.com/DioxusLabs/dioxus/pull/2171 and it also made me think and opened https://github.com/DioxusLabs/dioxus-std/issues/27

**Review by marc2332 — APPROVED** [review]

**Review by marc2332 — CHANGES_REQUESTED** [review]
image

Desktop example doesn't seem to work 🤔

**Comment by marc2332** [comment]
Ah wait, I need to run some commands, let me try

**Comment by marc2332** [comment]
All good!

**Pull request merged**